File: connector/php/TestWookieService.php
<?php
/**
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 * limitations under the License.
 */

//test service
ini_set('display_errors', 1);
error_reporting(E_ALL &~ E_NOTICE);

require("WookieConnectorService.php");

$test = new WookieConnectorService("http://127.0.0.1:8080/wookie/", "TEST", "localhost_test", 'demo_2');
//setup different userName
$test->getUser()->setLoginName("demo_1");
//get all available widgets
$availableWidgets = $test->getAvailableWidgets();
//print_r($availableWidgets);

//create select menus
echo '<pre>';
echo '<form action="" method="GET">';
echo '<select name="widget_id">';
echo '<option value="">No widget selected</option>';

foreach($availableWidgets as $Widget) {
	unset($sel);

	if($Widget->getIdentifier() == $_GET['widget_id']) {
		$sel = 'selected="selected"';
	}
	echo '<option value="'.$Widget->getIdentifier().'" '.$sel.'>'.$Widget->getTitle().'</option>';

}

echo '</select>';

//second select menu

echo '<select name="widget_id2">';
echo '<option value="">No widget selected</option>';

foreach($availableWidgets as $Widget) {
	unset($sel);

	if($Widget->getIdentifier() == $_GET['widget_id2']) {
		$sel = 'selected="selected"';
	}
	echo '<option value="'.$Widget->getIdentifier().'" '.$sel.'>'.$Widget->getTitle().'</option>';
}

echo '</select>';
echo '<input type="submit" value="Select" />';
echo '</form>';

if($_GET['widget_id'] != '') {
	//setup different userName for current user
	$test->getUser()->setLoginName("demo_2");
	$widget = $test->getOrCreateInstance($_GET['widget_id']);
	echo '<iframe src="'.$widget->getUrl().'" width="'.$widget->getWidth().'" height="'.$widget->getHeight().'"></iframe><br />';
	
	//add participant
	$testUser = new User('demo_2', 'demo_2');
	$test->addParticipant($widget, $testUser);
	print_r($test->getUsers($widget));
}

if($_GET['widget_id2'] != '') {
	//setup different userName for current user
	$test->getUser()->setLoginName("demo_3");
	$widget2 = $test->getOrCreateInstance($_GET['widget_id2']);
	echo '<iframe src="'.$widget2->getUrl().'" width="'.$widget2->getWidth().'" height="'.$widget2->getHeight().'"></iframe><br />';
	
	//add participant
	$testUser = new User('demo_3', 'demo_3');
	$test->addParticipant($widget2, $testUser);
	print_r($test->getUsers($widget2));
}

//call WidgetInstances->get <-- after widgets has been initialized

print_r($test->WidgetInstances->get());


?>

File: connector/python/TestWookieService.py
#
#  Licensed under the Apache License, Version 2.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
# limitations under the License.
#
from wookie import WookieConnectorService
from wookie.widget import Property
from wookie.widget import User

# @param host, path, api_key, shareddatakey
WookieConn = WookieConnectorService.WookieConnectorService('localhost:8080', '/wookie', 'TEST','localhost_python')
# set Widget user
WookieConn.setUser("demo_python", "demo_python_screenName")

print WookieConn.getCurrentUser().getLoginName()
print WookieConn.getCurrentUser().getScreenName()

# params: name, value, isPublic (default: false)
prop1 = Property.Property("test", "proov")
print prop1.getName()+":"+prop1.getValue()+":"+prop1.getIsPublic()

# get available widgets
widgetList = WookieConn.getAvailableWidgets()

a = widgetList[12]
b = widgetList[11]
#a = 'http://www.getwookie.org/widgets/weather'   

# create or get instances

retrievedInstance = WookieConn.getOrCreateInstance(a)
if retrievedInstance:
    print retrievedInstance.getTitle()+'\n'+retrievedInstance.getUrl()
    print retrievedInstance.getWidth()+'\n'+retrievedInstance.getHeight()

    # add participant, returns string "true" (if exists, or created) or "false"
    print WookieConn.addParticipant(retrievedInstance, User.User("ants_python", "ants_screenName"))
    
    # get participants, return list of participants
    users = WookieConn.getParticipants(retrievedInstance)
    print users[0].getLoginName()+":"+users[0].getScreenName()+":"+users[0].getThumbnail()

    # delete participant
    WookieConn.deleteParticipant(retrievedInstance, User.User("ants_python"));

    # add property
    # params: name, value, is_public
    prop2 = Property.Property("python_prop", "demo")
    WookieConn.setProperty(retrievedInstance, prop2)

    # get property, return property objec, if failed then "false"
    retrievedProp = WookieConn.getProperty(retrievedInstance, prop2)
    print retrievedProp.getValue()
    
    #delete property, returns "true" or "false"
    WookieConn.deleteProperty(retrievedInstance, prop2)

retrievedInstance2 = WookieConn.getOrCreateInstance(b)
if retrievedInstance2:
    print retrievedInstance2.getTitle()+'\n'+retrievedInstance2.getUrl()
    print retrievedInstance2.getWidth()+'\n'+retrievedInstance2.getHeight()


retrievedInstance3 = WookieConn.getOrCreateInstance('http://www.getwookie.org/widgets/weather')
if retrievedInstance3:
    print retrievedInstance3.getTitle()+'\n'+retrievedInstance3.getUrl()
    print retrievedInstance3.getWidth()+'\n'+retrievedInstance3.getHeight()
